File: javascript/two_sum.js
function twoSum(nums, target) {
    indicies = {}
    i = 0
    for (i; i <= nums.length; i++) {
        difference = target - nums[i]
        if (difference in indicies) {
            return [indicies[difference], i]
        }
        indicies[nums[i]] = i
    }
}

nums = [4, 7, 1, 8, 0]
target = 15

console.log(twoSum(nums, target))

File: python/two_sum.py
class Solution:
    def twoSum(self, nums, target):
        indices = {}
        for i in range(len(nums)):
            difference = target - nums[i]
            if difference in indices:
                return [indices[difference], i]
            indices[nums[i]] = i


trial = Solution()
nums = [4, 7, 1, 8, 0]
target = 15
print(trial.twoSum(nums=nums, target=target))
